### ISSUE TITLE
Add additional way of saving files

### ISSUE THREAD

**Issue by AlexisWalravens** [directus/api#736]
# Feature Request
Uploaded files should belong to records and not the opposite.
So when deleting a record, it should remove all uploaded files and thumbs for this record.

## What problem does this feature solve?
I don't like the fact that when I upload an image for example, I have to upload it to a big folder where every files ever uploaded are, and if I delete this record all the uploaded files linked to this record are not deleted.  

For example:  
If I create a products collection it makes no sense to store every pictures of each product in the same folder, and it makes no sense to keep these files if I delete a product.  
This way of handling file is good for a blog, in which a picture could be reused multiple times on different posts.

To be clear I don't want the current way of uploading file to be gone, I just want an alternative that I describe below.

## How do you think this should be implemented?
I'm used to work with October CMS. In October CMS there is two way of uploading a file/image:  
* Media Finder
* File Upload

The way Directus works is more similar to the Media Finder way, when uploading a file to a record, the id of the file is stored in a column of the record and the file is stored in a folder with ALL the other files. 

Deleting this record has not effect on the file.

With "File Upload" way works like this:
When saving a record with an uploaded file, a unique id is generated (for example `5c3dc484ad92c420571660`), the 9 first characters of this unique ID is split in 3 to make a folder path where the file will be stored, like this: `uploads/5c3/dc4/84a/file.ext` (created thumbs for this file stores them in the same folder), then in the files table in the database, a new record is created with various file meta data AND the **name of the field**, the **model** and **ID** of the record it belongs to.

Deleting the record deletes the file record in the files table and removes all the files/images and thumbs.
So in Directus it would be something like storing the field name, the collection ID and the ID of the record in the files table.

## Would you be willing to work on this?
Unfortunately I don't have time for the moment, but I really wanted to post this, as I like the way Directus works but this is the main thing that prevent me to use it. (and the lack of a repeater field but I read it is on the way).

**Comment by rijkvanzanten** [comment]
TL;DR

Add a way of saving files that ties the file to a row in a collection

**Comment by benhaynes** [comment]
To achieve better clarity/visibility, we are now tracking feature requests within the [Feature Request](https://github.com/directus/api/projects/40) project board.

**This issue being closed does not mean it's not being considered.**
